projects/packages/forms/src/blocks/field-name/edit.js

```diff
@@ -1,12 +1,42 @@
+import { useEffect } from '@wordpress/element';
 import { __ } from '@wordpress/i18n';
 import JetpackField from '../shared/components/jetpack-field.js';
 import useFormWrapper from '../shared/hooks/use-form-wrapper.js';
-import useNameLabelSync from './hooks/use-name-label-sync.js';
+import useNameFieldTransforms from './hooks/use-name-field-transforms.js';
+import {
+	isFirstNameVariationId,
+	isLastNameVariationId,
+	isNameVariationId,
+	FIRST_NAME_ID,
+	LAST_NAME_ID,
+	NAME_ID,
+} from './variations.js';
 
 export default function NameFieldEdit( props ) {
+	const { clientId, attributes, setAttributes } = props;
+	const { id, fieldVariant } = attributes || {};
+
 	useFormWrapper( props );
 
-	useNameLabelSync( { clientId: props.clientId, id: props.attributes?.id } );
+	// Initialize fieldVariant for backward compatibility with existing Name field blocks.
+	useEffect( () => {
+		if ( fieldVariant ) {
+			return;
+		}
+		if ( isFirstNameVariationId( id ) ) {
+			setAttributes( { fieldVariant: FIRST_NAME_ID } );
+		} else if ( isLastNameVariationId( id ) ) {
+			setAttributes( { fieldVariant: LAST_NAME_ID } );
+		} else if ( isNameVariationId( id ) ) {
+			setAttributes( { fieldVariant: NAME_ID } );
+		} else {
+			// Default to base 'name' variant for any other case (empty id or 'name' id)
+			setAttributes( { fieldVariant: '' } );
+		}
+	}, [ clientId, fieldVariant, id, setAttributes ] );
```

Comment:
I added a bit of backwards compatibility support by trying to set the new fieldVariant attribute based on the HTML ID.